rfc#00003

```diff
@@ -0,0 +1,213 @@
+
+Note - there are models where a trust relationship is maintained by
+the tenant to the Tenant workload image repository or to the tenant
+workload, but it is not directly within the trust domain.  These
+models are not examined in this document.
+
+#### State 1 non-Enarx-enabled use case (provisioned)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Tenant workload
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+The change from state 0 to 1 is the provisioning of the Tenant workload
+onto the host from the Tenant workload image.
+
+The key point of this use case from a trust domain point of view it that
+since the host has access to all processes and associated memory that are
+running on it, all workloads that run on the host can only exist within
+the host trust domain.
+
+### Enarx use case states
+Note that exact state details may vary by implementation.
+
+#### State 0 Enarx use case (initial state)
+Initial state (we assume no compromised processes on host):
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Enarx runtime image
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Enarx runtime image
+  - Tenant workload image
+  - Tenant workload image repository
+This is the same as the initial state for the non-Enarx-enabled
+use case.
+
+#### State 1 Enarx use case (TODO: check name - pre-attestation)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - TEE instance
+  - Enarx runtime image (within TEE instance)
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Tenant workload image
+  - Tenant workload image repository
+The state change from 0 to one creates a TEE instance and loads
+the Enarx runtime image into it.
+
+#### State 2 Enarx use case (TODO: check name - attested)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Empty Keep
+  - Tenant workload image
+  - Tenant workload image repository
+The state change from 1 to 2 moves the Empty Keep into the tenant
+trust domain.
+
+#### State 3 Enarx use case (TODO: check name - provisioned)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Provisioned Keep
+  - Tenant workload image
+  - Tenant workload image repository
+The state change from 2 to 3 provisions the tenant workload into
+the (previously empty) Keep.
+
+
+### Applications and trust domains in Enarx
+It may be that the tenant workload, once running, should not
+continue to exist in the top-level tenant trust domain, but may
+be moved into a sub-domain, or even transferred to another owner or
```

Comment:
> may be moved into a sub-domain

Why?

Reply:
Because you may have a hierarchy of trust domains.

Reply:
:+1: